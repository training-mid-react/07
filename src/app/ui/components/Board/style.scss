.board {
    display: flex;
    justify-content: center;
    background-color: transparent;
    padding-top: 50px;
    background-color: transparent;
}


