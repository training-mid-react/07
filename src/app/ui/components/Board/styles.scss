.board {
    display: flex;
    justify-content: center;
    gap: 10px;
    width: fit-content;
    margin: auto;
    padding: 1rem;
    border-radius: 10px;
    background-color: #5454564b;
    height: 100%;
    box-shadow: 10px 10px 5px rgba(0, 0, 0, 0.221);

    &__column {
        display: flex;
        flex-direction: column;
        gap: 10px;

        &:hover {
            position: relative;

            &::before {
                position: absolute;
                top: -1rem;
                content: '';
                width: 0;
                height: 0;
                border-top: 25px solid black;
                border-right: 25px solid transparent;
                border-left: 25px solid transparent;
                border-bottom: 0;
            }
        }
    }

    .falling {
        animation: fall 0.8s ease-out;
    }

    @keyframes fall {
        from {
            transform: translateY(-350px);
        }
        to {
            transform: translateY(0);
        }
    }
}
