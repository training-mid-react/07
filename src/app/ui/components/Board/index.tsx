import { FC } from 'react';
import { Column } from '../Column';
import { IBoard } from '../../../../core/interfaces';
import './style.scss';

interface Props {
  board: IBoard;
  handleDrop: (colIndex: number) => void;
}

export const Board:FC<Props> = ({board, handleDrop}) => {

  return (
    <div className="board">
      {board.map((_, colIndex) => (
        <Column key={colIndex} board={board} columnIndex={colIndex} handleDrop={handleDrop} />
      ))}
      <Column key={board.length} board={board} columnIndex={board.length} handleDrop={handleDrop} />
    </div>
  );
};
