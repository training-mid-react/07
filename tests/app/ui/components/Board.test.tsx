import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Board from '../../../../src/app/ui/components/Board';
import { describe, expect, it, vi } from 'vitest';
import '@testing-library/jest-dom';

describe('Pruebas en el componente Board', () => {
  it('debe coincidir con el snapshot', () => {
    const mockBoard = [
      [0, 0, 0, 0, 0, 0, 0],
      [0, 0, 0, 0, 0, 0, 0],
      [0, 0, 0, 0, 0, 0, 0],
      [0, 0, 0, 0, 0, 0, 0],
      [0, 0, 0, 0, 0, 0, 0],
      [0, 0, 0, 0, 0, 0, 0],
    ];

    render(<Board board={mockBoard} handleDropPiece={() => {}} />);
    expect(document.body).toMatchSnapshot();
  });
  it('debe llamar al handleDropPiece al hacer clic en la columna', () => {
    const mockBoard = [
      [0, 0, 0, 0, 0, 0, 0],
      [0, 0, 0, 0, 0, 0, 0],
      [0, 0, 0, 0, 0, 0, 0],
      [0, 0, 0, 0, 0, 0, 0],
      [0, 0, 0, 0, 0, 0, 0],
      [0, 0, 0, 0, 0, 0, 0],
    ];

    const handleDropPiece = vi.fn();
    const { getByTestId } = render(
      <Board board={mockBoard} handleDropPiece={handleDropPiece} />
    );

    const column = getByTestId('column-0');
    expect(column).toBeInTheDocument();

    fireEvent.click(column);

    expect(handleDropPiece).toHaveBeenCalled(); 
  });

  it('debe llamar al handleDropPiece con el Ã­ndice correcto', () => {
    const mockBoard = [
      [0, 0, 0, 0, 0, 0, 0],
      [0, 0, 0, 0, 0, 0, 0],
      [0, 0, 0, 0, 0, 0, 0],
      [0, 0, 0, 0, 0, 0, 0],
      [0, 0, 0, 0, 0, 0, 0],
      [0, 0, 0, 0, 0, 0, 0],
    ];

    const handleDropPiece = vi.fn();
    const { getByTestId } = render(
      <Board board={mockBoard} handleDropPiece={handleDropPiece} />
    );

    const column = getByTestId('column-0');
    fireEvent.click(column);

    expect(handleDropPiece).toHaveBeenCalledWith(0);
  });
});
