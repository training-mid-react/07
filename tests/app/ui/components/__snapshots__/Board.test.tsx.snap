// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pruebas en el componente Board > debe coincidir con el snapshot 1`] = `
<body>
  <div>
    <div
      class="board"
    >
      <div
        class="column"
        data-testid="column-0"
      >
        <div
          class="cell  "
          data-testid="cell-0-0"
        />
        <div
          class="cell  "
          data-testid="cell-0-1"
        />
        <div
          class="cell  "
          data-testid="cell-0-2"
        />
        <div
          class="cell  "
          data-testid="cell-0-3"
        />
        <div
          class="cell  "
          data-testid="cell-0-4"
        />
        <div
          class="cell  "
          data-testid="cell-0-5"
        />
      </div>
      <div
        class="column"
        data-testid="column-1"
      >
        <div
          class="cell  "
          data-testid="cell-1-0"
        />
        <div
          class="cell  "
          data-testid="cell-1-1"
        />
        <div
          class="cell  "
          data-testid="cell-1-2"
        />
        <div
          class="cell  "
          data-testid="cell-1-3"
        />
        <div
          class="cell  "
          data-testid="cell-1-4"
        />
        <div
          class="cell  "
          data-testid="cell-1-5"
        />
      </div>
      <div
        class="column"
        data-testid="column-2"
      >
        <div
          class="cell  "
          data-testid="cell-2-0"
        />
        <div
          class="cell  "
          data-testid="cell-2-1"
        />
        <div
          class="cell  "
          data-testid="cell-2-2"
        />
        <div
          class="cell  "
          data-testid="cell-2-3"
        />
        <div
          class="cell  "
          data-testid="cell-2-4"
        />
        <div
          class="cell  "
          data-testid="cell-2-5"
        />
      </div>
      <div
        class="column"
        data-testid="column-3"
      >
        <div
          class="cell  "
          data-testid="cell-3-0"
        />
        <div
          class="cell  "
          data-testid="cell-3-1"
        />
        <div
          class="cell  "
          data-testid="cell-3-2"
        />
        <div
          class="cell  "
          data-testid="cell-3-3"
        />
        <div
          class="cell  "
          data-testid="cell-3-4"
        />
        <div
          class="cell  "
          data-testid="cell-3-5"
        />
      </div>
      <div
        class="column"
        data-testid="column-4"
      >
        <div
          class="cell  "
          data-testid="cell-4-0"
        />
        <div
          class="cell  "
          data-testid="cell-4-1"
        />
        <div
          class="cell  "
          data-testid="cell-4-2"
        />
        <div
          class="cell  "
          data-testid="cell-4-3"
        />
        <div
          class="cell  "
          data-testid="cell-4-4"
        />
        <div
          class="cell  "
          data-testid="cell-4-5"
        />
      </div>
      <div
        class="column"
        data-testid="column-5"
      >
        <div
          class="cell  "
          data-testid="cell-5-0"
        />
        <div
          class="cell  "
          data-testid="cell-5-1"
        />
        <div
          class="cell  "
          data-testid="cell-5-2"
        />
        <div
          class="cell  "
          data-testid="cell-5-3"
        />
        <div
          class="cell  "
          data-testid="cell-5-4"
        />
        <div
          class="cell  "
          data-testid="cell-5-5"
        />
      </div>
      <div
        class="column"
        data-testid="column-6"
      >
        <div
          class="cell  "
          data-testid="cell-6-0"
        />
        <div
          class="cell  "
          data-testid="cell-6-1"
        />
        <div
          class="cell  "
          data-testid="cell-6-2"
        />
        <div
          class="cell  "
          data-testid="cell-6-3"
        />
        <div
          class="cell  "
          data-testid="cell-6-4"
        />
        <div
          class="cell  "
          data-testid="cell-6-5"
        />
      </div>
    </div>
  </div>
</body>
`;
